### DISCUSSION TITLE
Dockerization?

### DISCUSSION THREAD

**Discussion by cabbey** [broofa/flightcard#12] [General]
Any interest in Dockerization of this code base? I see a couple ways this could help based on some of the discussions and issues and todos:

1. Test and dev databases - if we do it right, everyone that wants to hack on it can spin up a local database to work against. Snapshots or scripted fixtures can be shared, or a database on the net can be shared by multiple devs.

2. Deployments in an offline location - a single laptop sitting near LCO/RSO area could host the full app and a WiFi network that clients could connect to for example. Or a small dedicated server and a WAP.

(I was unable to make the meeting, so have no context beyond the mention on the list and what I've seen here on GitHub, but if this is even half as serious as it looks, I'm happy to thrown in to help where I can with backend infrastructure and such as that's my day job, and my wife has already volunteered her QA skills, again her day job, even if it means time away from her L1 practice.)

**Comment by broofa** [comment]
Hi @cabbey, great to have you here!

Regarding Docker, I have a few reservations about this...

1. I don't think Firebase (which I'm using for auth and realtime datastore) is setup for Docker. ([source](https://www.reddit.com/r/Firebase/comments/gocgav/run_firebase_locally_in_a_docker_container/)). 
2. Running an on-site server solves the connectivity problem (sort of, see below) but comes at the cost of having to drop features that you get with cloud-connectivity.  E.g. Access to the thrustcurve.org API, or having access to flight cards post-launch.  (I know there are solutions / workarounds - e.g. some sort of data sync feature - but the cost to implement is higher.)
3. Setting up site-wide WIFI is non-trivial.  "_What hardware is needed? How is it powered?  What's the effective range?  What if your device actually has good internet coverage - will that continue to work while on the launch site LAN? etc..._"). 
4. Last, and most humbling, is that I've never been particularly comfortable working in Docker.  The couple of times I've used it, it's felt like a pretty big barrier to rapid, iterative, development.  That's just a personal hangup, though, which I'd be happy to put some effort into figuring out.

Curious what your thoughts are.
